apps/frontend/src/app/poll/choose-events/choose-events.component.ts

```diff
@@ -202,13 +210,32 @@ export class ChooseEventsComponent implements OnInit {
     } else if (maxParticipants && this.participants.length >= maxParticipants) {
       this.closedReason = 'This poll has reached it\'s maximum number of participants. You can no longer submit your vote.';
       this.showResults = true;
-    } else {
-      this.closedReason = undefined;
-      this.showResults = !this.poll?.settings?.blindParticipation || this.isAdmin || this.userVoted();
     }
   }
 
-  private clearSelection(){
+  private updateHiddenReason() {
+    this.showResults = this.isAdmin || this.userVoted() ||
+      this.poll?.settings?.showResult === ShowResultOptions.IMMEDIATELY;
+
+    switch (this.poll?.settings.showResult) {
+      case ShowResultOptions.NEVER:
+        this.showNever = !this.userVoted() && !this.isAdmin;
+        if (!this.isAdmin) {
+          this.hiddenReason = 'The results of this poll are hidden. You will only be able to see your own votes.';
+        }
+        this.showResults = true;
+        break;
+      case ShowResultOptions.AFTER_PARTICIPATING:
+        if (!this.showResults) {
+          this.hiddenReason = 'This is a blind poll. You can\'t see results or other user\'s votes until you participate yourself.';
+        }
```

Comment:
Needs else this.hiddenReason = undefined

Reply:
Fixed in f85dcf7.